cheatGUI/src/utils/wasd.ts

```diff
@@ -0,0 +1,57 @@
+import { useWASD } from "../hacks/location";
+
+export enum WASD {
+    None,
+    Normal,
+    Phasing,
+}
+
+
+function ChangeX (x : number) {
+    for (let i = 0; i < 100; i++) {
+        setTimeout(() => {  _.player._playerContainer.x -= x/100; }, 50);
+    }
+}
+
+function ChangeY (y : number) {
+    for (let i = 0; i < 100; i++) {
+        setTimeout(() => {  _.player._playerContainer.y -= y/100; }, 50);
```

Comment:
Why setTimeout?

Reply:
What do you suggest we use?

Reply:
What's wrong with setTimeout?

Reply:
Do you even know what it does? There is no point in doing something like that.

Reply:
lmao yes, it runs a function after a delay,

Reply:
What you are doing is the same as
```ts
function ChangeY (y : number) {
    setTimeout(() => {
        for (let i = 0; i < 100; i++) {
            _.player._playerContainer.y -= y/100;
    }}, 50)
}
```

Reply:
Tis not the same, cuz we are doing it inside the loop, not outside

Reply:
🍿

Reply:
**am i being sped**

Reply:
Probably, but how would I know?

Reply:
alright we can use async/await for those functions, and put await before literally **everything** in that function, also use some form of proper wait?